lib/analyzer/os-release-detector.ts

```diff
@@ -0,0 +1,139 @@
+import subProcess from '../sub-process';
+
+export {
+  detect,
+};
+
+async function dockerCat(targetImage, file): Promise<string> {
+  try {
+    return await subProcess.execute('docker', [
+      'run', '--rm', targetImage, 'cat', file,
+    ]);
+  } catch (stderr) {
+    if (typeof stderr === 'string' && stderr.indexOf('No such file') >= 0) {
+      return '';
+    }
+    throw new Error(stderr);
+  }
+}
+
+async function detect(targetImage) {
+  let osRelease = await tryOSRelease(targetImage);
+
+  // First generic fallback
+  if (!osRelease) {
+    osRelease = await tryLSBRelease(targetImage);
+  }
+
+  // Fallbacks for specific older distributions
+  if (!osRelease) {
+    osRelease = await tryDebianVersion(targetImage);
+  }
+
+  if (!osRelease) {
+    osRelease = await tryAlpineRelease(targetImage);
+  }
+
+  if (!osRelease) {
+    osRelease = await tryOracleRelease(targetImage);
+  }
+
+  if (!osRelease) {
+    osRelease = await tryRedHatRelease(targetImage);
+  }
+
+  if (!osRelease) {
+    throw new Error('Failed to detect OS release');
+  }
+
+  // Oracle Linux identifies itself as "ol"
+  if (osRelease.name === 'ol') {
+    osRelease.name = 'oracle';
+  }
+
+  return osRelease;
+}
+
+async function tryOSRelease(targetImage) {
+  const text = await dockerCat(targetImage, '/etc/os-release');
+  if (!text) {
+    return null;
+  }
+  const idRes = text.match(/^ID=(.+)$/m);
+  if (!idRes) {
+    throw new Error('Failed to parse /etc/os-release');
+  }
+  const name = idRes[1].replace(/"/g, '');
```

Comment:
maybe extract ".replace(/"/g, '');" to a function - you are using it multiple times

Reply:
I'd rather leave them inline so that the code can be read easily